docs/src/pages/components/data-grid/demo/FullFeauredDemo.tsx

```diff
@@ -0,0 +1,304 @@
+          <MenuItem value={25}>25</MenuItem>
+          <MenuItem value={100}>100</MenuItem>
+          <MenuItem value={1000}>{Number(1000).toLocaleString()}</MenuItem>
+        </Select>
+      </FormControl>
+      <FormControl component="fieldset">
+        <InputLabel>Theme</InputLabel>
+        <Select value={activeTheme} onChange={handleThemeChange}>
+          <MenuItem value="default">Default Theme</MenuItem>
+          <MenuItem value="ant">Ant Design</MenuItem>
+        </Select>
+      </FormControl>
+      <Button
+        size="small"
+        variant="outlined"
+        color="primary"
+        onClick={handleApplyChanges}
+      >
+        <KeyboardArrowRightIcon fontSize="small" /> Apply
+      </Button>
+    </FormGroup>
+  );
+}
+
+export default function FullFeaturedDemo() {
+  const classes = useStyles();
+  const AntDesignClasses = useStylesAntDesign();
+  const [isAntDesign, setIsAntDesign] = React.useState<boolean>(false);
+  const [type, setType] = React.useState<GridDataType>('Commodity');
+  const [size, setSize] = React.useState(100);
+  const { data, setRowLength, loadNewData } = useDemoData({
+    dataSet: type,
+    rowLength: size,
+    maxColumns: 20,
+  });
+
+  const [pagination, setPagination] = React.useState<GridPaginationSettings>({
+    pagination: false,
+    autoPageSize: false,
+    pageSize: undefined,
+  });
+
+  const getActiveTheme = () => {
+    return isAntDesign ? 'ant' : 'default';
+  };
+
+  const handleApplyClick = (settings) => {
+    if (size !== settings.size) {
+      setSize(settings.size);
+    }
+
+    if (type !== settings.type) {
+      setType(settings.type.toLowerCase());
+    }
+
+    if (getActiveTheme() !== settings.theme) {
+      setIsAntDesign(!isAntDesign);
+    }
+
+    if (size !== settings.size || type !== settings.type) {
+      setRowLength(settings.size);
+      loadNewData();
+    }
+
+    const newPaginationSettings: GridPaginationSettings = {
+      pagination: settings.pagesize !== -1,
+      autoPageSize: settings.pagesize === 0,
+      pageSize: settings.pagesize > 0 ? settings.pagesize : undefined,
+    };
+
+    setPagination(
+      (
+        currentPaginationSettings: GridPaginationSettings,
+      ): GridPaginationSettings => {
+        if (
+          currentPaginationSettings.pagination ===
+            newPaginationSettings.pagination &&
+          currentPaginationSettings.autoPageSize ===
+            newPaginationSettings.autoPageSize &&
+          currentPaginationSettings.pageSize === newPaginationSettings.pageSize
+        ) {
+          return currentPaginationSettings;
+        }
+        return newPaginationSettings;
+      },
+    );
+  };
+
+  return (
+    <div className={classes.root}>
+      <SettingsPanel
+        onApply={handleApplyClick}
+        size={size}
+        type={type}
+        theme={getActiveTheme()}
+      />
+      <XGrid
+        className={isAntDesign ? AntDesignClasses.root : undefined}
```

Comment:
Interesting, if we want to push the approach further. We would need to use the `ThemeProvider`. For instance, would allow to customize the component deeper, like the typography or the button styles

<img width="442" alt="Screenshot 2021-03-05 at 15 19 59" src="https://user-images.githubusercontent.com/3165635/110127773-5059ef80-7dc6-11eb-9c79-58d4aedd714f.png">

@dtassone any preference for when we take on this problem? I think that it could be later, but no strong preferences.

Reply:
I agree, ideally should be with the theme provider but nailing the styles down would take more time.

Reply:
@oliviertassinari let's create a separate issue on prioritize in the next planning session :)